doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,204 @@
+:func:`~.QNode` constructor function to create the specific type of qnode based on
+the device, interface, and quantum function.
+
+A widespread representation of quantum circuits is by creating a `Directed
+Acyclic Graph (DAG)
+<https://pennylane.ai/qml/glossary/hybrid_computation.html#directed-acyclic-graphs>`__
+and representing quantum operations within such a graph. Each ``QNode``
+represents the quantum circuit by building such a DAG by creating a
+:class:`~.CircuitGraph` instance.
+
+For further details on QNodes, and a full list of QNodes, refer to the
+:doc:`/code/qml_qnodes` module.
+
+Interfaces
+**********
+
+The integration between classical and quantum computations is encompassed by
+interfaces. Differentiable and Jacobian ``QNodes`` have interfacing functions
+that provide a 'wrapper' around QNodes such. These wrappers further transform
+the ``QNode`` such that the quantum gradient rules of the QNodes are registered
+to the machine learning interface via a custom gradient class or function.
+
+We refer to the :ref:`intro_interfaces` page for a more in-depth introduction
+and a list of available interfaces.
+
+Key design details
+##################
+
+The following are key design details related to how PennyLane works internally.
+
+Operators
+*********
+
+Quantum operators are incorporated by the :class:`~.Operator` class which
+contains basic information about the operator (e.g. number of parameters,
+number of wires it acts on, etc.) and further convenience methods (e.g.
+:attr:`~.Operator.matrix`, :attr:`~.Operator.eigvals`.
+
+Two important subclasses of the ``Operator`` class are:
+
+* the :class:`~.Operation` class representing quantum gates and
+* the :class:`~.Observable` representing quantum observables specified for
+  measurement.
+
+Together with ``Operator``, these classes serve as base classes for quantum
+operators.
+
+Certain operators can serve as both quantum gates and observables (e.g.
+:class:`~.PauliZ`, :class:`~.PauliX`, etc.). Such classes inherit from both
+``Operation`` and ``Observable`` classes.
+
+Quantum operator instances, in general, are used to build quantum functions
+which are used within ``QNode`` instances. Users can define such quantum
+functions by creating regular Python functions and incorporating ``Operator``
+instances.
+
+The following is an example of this using the :func:`~.qnode` decorator and a
+valid pre-defined device (``dev``).
+
+.. code-block:: python
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.PauliX(0)
+        return qml.expval(qml.PauliZ(0))
+
+This syntax of PennyLane results in the Operator instances not being "recorded"
+by the Python function. What is more, there is actually *no composition logic*
+in between the ``QNode`` that is being created and the ``Operator`` instances
+(``qml.PauliX(0)`` and ``qml.PauliZ(0)``), as there are merely class
+instantiations happening upon calling the ``circuit`` quantum function.
```

Comment:
This paragraph is a bit unclear to me 🤔

Reply:
I've reworded it a bit, do you think it's better or would there be further things that are unclear?